### PR TITLE
FUSETOOLS-2936 - upgrade to Eclipse Photon TP

### PR DESCRIPTION
**Pull request by lhein**
Signed-off-by: Lars Heinemann <lhein.smx@gmail.com>

# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [x] Did you use the Jira Issue number in the commit comments?
- [x] Did you set meaningful commit comments on each commit?
- [x] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [ ] Did the CI job report a successful build?
- [ ] Is the non-happy path working, too?

## Maintainability

- [ ] Are all Sonar reported issues fixed or can they be ignored?
- [ ] Is there no duplicated code?
- [ ] Are method-/class-/variable-names meaningful?

## Tests

- [ ] Are there unit-tests?
- [ ] Are there integration tests (or at least a jira to tackle these)?
- [ ] Do we need a new UI test?

## Legal

- [x] Have you used the correct file header copyright comment?
- [x] Have you used the correct year in the headers of new files?

### CODE REVIEW

**Comment by lhein** [comment]
this build will fail because the photon update site for JBT is currently empty

**Comment by lhein** [comment]
testPR

**Comment by lhein** [comment]
still some issues to fix:

- [ ] update the target file with latest photon versions
- [ ] fix compile issues due to API changes

**Comment by apodhrad** [comment]
Hi @lhein, please include the following PR to your existing one
https://github.com/jbosstools/jbosstools-fuse/pull/1356/files

**Comment by lhein** [comment]
@apodhrad I don't think that is a good idea. I saw several things in that commit that are either wrong or no longer required. Reddeer 2.1 for instance is now in the TP. The gethandleID method impl is badly implemented. I'd rather suggest to redo that PR.

**Comment by bfitzpat** [comment]
Built successfully here. All is good!

**Comment by bfitzpat** [comment]
Rebuilding after the red deer test update...

File: core/plugins/org.fusesource.ide.foundation.ui/src/org/fusesource/ide/foundation/ui/views/TabFolderSupport2.java [hunk 1]
```diff
@@ -70,6 +70,11 @@ public String getContributorId() {
 				if (configurationElements == null || configurationElements.length == 0) {
 					configurationElements = new IConfigurationElement[] { new IConfigurationElement() {
 
+						@Override
+						public int getHandleId() {
+							return contributorId.hashCode();
+						};
```
[the remark this answers is not in the document]
Comment (lhein) [reply core/plugins/org.fusesource.ide.foundation.ui/src/org/fusesource/ide/foundation/ui/views/TabFolderSupport2.java]: false positive

**Comment by lhein** [comment]
merged as previous build succeeded and I removed just a ; char which was too much.

**Pull request merged**